### PR TITLE
wireshark: 3.6.5 -> 4.0.1

### PR DESCRIPTION
**Pull request by esclear**
###### Description of changes

The release notes for the major update to 4.0.0 are at https://www.wireshark.org/news/20221004.html

The release notes for the 4.0.1 release can be found at https://www.wireshark.org/docs/relnotes/wireshark-4.0.1.html

This is a draft at the moment, as I want to test some more.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by esclear** [comment]
While trying to rebuild all dependent packages, I noticed that the `pyshark` python modules fail to build (and thus also `credslayer`, which depends on `pyshark`).

The problem is caused by failing tests, which in turn are caused by a change in the data format returned by wireshark for a command, the output of which is parsed by `pyshark`.
I proposed a fix for this in KimiNewt/pyshark#608.

I think that `credslayer` might be broken on `unstable` already.

**Comment by gador** [comment]
@esclear so do you think it is worth waiting for the upstream fix for `pyshark` before updating `wireshark`?

**Comment by esclear** [comment]
@gador I'm not sure, especially since I don't know when it might get merged.

Personally, I would really like to see Wireshark 4 in nixpkgs and have considered adding a patch to the pyshark derivation, at least until the fix got upstream.
There is only one nix package that depends on pyshark, namely credslayer. But as pyshark seems to be used a bunch, I would like to keep a working build in nixpkgs.

**Comment by gador** [comment]
maybe we should add an override to `pyshark` to use the supported `wireshark` version and add a comment and a link to your PR? This way we could update the `wireshark` nixpkgs-wide and `pyshark` would continue to work and can use the new version when upstream supports it.

**Comment by esclear** [comment]
> maybe we should add an override to `pyshark` to use the supported `wireshark` version and add a comment and a link to your PR? This way we could update the `wireshark` nixpkgs-wide and `pyshark` would continue to work and can use the new version when upstream supports it.

I'm not quite sure what you are proposing exactly.
Wouldn't we need to keep a second wireshark package around to use as an override for that?
Or can we override the wireshark `src`?

**Comment by gador** [comment]
hi @esclear,
I did actually overwrite the `src` of wireshark first in `pyshark`. This would've been my suggestion, too. Then I saw you created a PR with a fix upstream, so I just applied it.
Please see https://github.com/NixOS/nixpkgs/commit/052b3b3d96e7ee9ce54e9f1d4d17c7a22793e14a

**Comment by gador** [comment]
> I think that `credslayer` might be broken on `unstable` already.

I marked it as `broken`: https://github.com/NixOS/nixpkgs/pull/198909

File: pkgs/applications/networking/sniffers/wireshark/default.nix [hunk 1]
```diff
@@ -44,7 +44,7 @@ in stdenv.mkDerivation {
   depsBuildBuild = [ buildPackages.stdenv.cc ];
 
   buildInputs = [
-    gettext pcre libpcap lua5 libssh nghttp2 openssl libgcrypt
+    gettext pcre pcre2 libpcap lua5 libssh nghttp2 openssl libgcrypt
```
Comment (fpletz) [inline pkgs/applications/networking/sniffers/wireshark/default.nix:47]: Seems like pcre is not required anymore.

**Comment by LeSuisse** [comment]
Flagging as security related, this upgrade contains a fix for CVE-2022-3725

https://www.wireshark.org/security/wnpa-sec-2022-07.html

**Comment by esclear** [comment]
@LeSuisse Good catch! Hope that this PR can get merged soon, then.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64. WFM macos 10.15.

**Comment by teto** [comment]
Passed nixpkgs-review, I rebuilt the system with it and the .desktop is picked up correctly. I could capture packets from wifi interface so all good.

**Comment by risicle** [comment]
We need to find the equivalent bump for the stable branch.

**Comment by LeSuisse** [comment]
The sec issues have been fixed in #199697 for 22.05.

**Pull request merged**